vllm/entrypoints/openai/serving_embedding.py

```diff
@@ -21,7 +23,16 @@
 
 logger = init_logger(__name__)
 
-TypeTokenIDs = List[int]
+
+def check_embedding_mode(model_config: ModelConfig) -> bool:
+    embedding_mode = model_config.task == "embedding"
+
+    if not embedding_mode:
+        logger.warning("embedding_mode is False. Embedding API will not work.")
```

Comment:
Shouldn't this be an error? 

or are you thinking an initialized model server can handle both completions and embedding generation? In that case, we should make `model_config.task` a list and check if `embedding` is in it, right?